### ISSUE TITLE
Subclassing fails on 2.1.0rc2

### ISSUE THREAD

**Issue by SaeX** [yourlabs/django-autocomplete-light#397]
If I do the following, then I get a proper autocomplete:

```
autocomplete_light.register(Author, search_fields=['name'],)

class BookForm(autocomplete_light.ModelForm):

    class Meta:
        model = models.Book
        fields = ('authors', )
```

However, when subclassing the form as in the example below, I no longer get the autocomplete:

```
autocomplete_light.register(Author, search_fields=['name'],)

class BookFormBase(autocomplete_light.ModelForm):

    class Meta:
        model = models.Book
        fields = ('authors', )


class BookForm(BookFormBase):
    pass  # for demo only, in reality e.g. some clean() method could be overridden
```

This used to work in 2.0.9 but fails in version 2.1.0rc2.

**Comment by blueyed** [comment]
That might be related to https://github.com/yourlabs/django-autocomplete-light/commit/2940d3bef369eba417b11426876b7946d822d189

**Comment by jpic** [comment]
Absolutely related

**Comment by jpic** [comment]
Could you define "fails" @SaeX  ?

Thanks !

**Comment by SaeX** [comment]
Sure @jpic. "Fails" as in "no autocomplete shown". A Select or SelectMultiple widget is shown instead (as if there was no django-autocomplete-light installed).

**Comment by jpic** [comment]
I can't reproduce your issue, maybe your autocomplete is not registered correctly anymore in 2.1.0rc2 ?

Could you check that the autocomplete was registered correctly ?

**Comment by jpic** [comment]
HINT: Try the autocomplete include root url as superuser, it'll display the registry.

**Comment by SaeX** [comment]
@jpic, I've created a repo with an example project to show what the issue is about. I was able to reproduce in this small project as well. 

Please see https://github.com/SaeX/dal_issue_397/ (and `forms.py` in particular). I get an autocomplete widget with 2.0.9, but are getting a default SelectMultiple widget with 2.1.0rc2.

**Comment by jpic** [comment]
It seems working for me on py34 django 18
![2015-04-11-111801_386x141_scrot](https://cloud.githubusercontent.com/assets/94636/7100905/50e5d94c-e03c-11e4-9381-d46b4ede94cc.png)

Could you describe the problem ?

Thanks !

**Comment by jpic** [comment]
Tried on a fresh venv too:

```
[env] 11/04 2015 12:40:26 jpic@lue /tmp 
$ git clone https://github.com/SaeX/dal_issue_397.git
Cloning into 'dal_issue_397'...
remote: Counting objects: 26, done.
remote: Compressing objects: 100% (23/23), done.
remote: Total 26 (delta 1), reused 22 (delta 0), pack-reused 0
Unpacking objects: 100% (26/26), done.
Checking connectivity... done.

[env] 11/04 2015 12:40:37 jpic@lue /tmp 
$ virtualenv dal_issue_env
Using real prefix '/usr'
New python executable in dal_issue_env/bin/python2
Also creating executable in dal_issue_env/bin/python
Installing setuptools, pip...done.

[env] 11/04 2015 12:40:44 jpic@lue /tmp 
$ source dal_issue_env/bin/activate

[dal_issue_env] 11/04 2015 12:40:48 jpic@lue /tmp 
$ pip install django django-autocomplete-light==2.1.0rc2
You are using pip version 6.0.8, however version 6.1.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
DEPRECATION: --download-cache has been deprecated and will be removed in the future. Pip now automatically uses and configures its cache.
Collecting django
  Using cached Django-1.8-py2.py3-none-any.whl
Collecting django-autocomplete-light==2.1.0rc2
  Downloading django-autocomplete-light-2.1.0rc2.tar.gz (95kB)
    4% |#                               | 4.1kB 38.3MB/s eta 0:00:0    100% |################################| 98kB 769kB/s 
Collecting six>=1.4 (from django-autocomplete-light==2.1.0rc2)
  Using cached six-1.9.0-py2.py3-none-any.whl
Installing collected packages: six, django-autocomplete-light, django

  Running setup.py install for django-autocomplete-light

Successfully installed django-1.8 django-autocomplete-light-2.1.0rc2 six-1.9.0

[dal_issue_env] 11/04 2015 12:41:04 jpic@lue /tmp 
$ cd dal_issue_397/

[dal_issue_env] 11/04 2015 12:41:20 jpic@lue /tmp/dal_issue_397 () 
$ python manage.py runserver
Performing system checks...

System check identified no issues (0 silenced).
April 11, 2015 - 10:41:23
Django version 1.8, using settings 'dal_issue_397.settings'
Starting development server at http://127.0.0.1:8000/
Quit the server with CONTROL-C.
/tmp/dal_issue_env/lib/python2.7/site-packages/taggit/models.py:85: RemovedInDjango19Warning: Model class taggit.models.Tag doesn't declare an explicit app_label and either isn't in an application in INSTALLED_APPS or else was imported before its application was loaded. This will no longer be supported in Django 1.9.
  class Tag(TagBase):

/tmp/dal_issue_env/lib/python2.7/site-packages/taggit/models.py:185: RemovedInDjango19Warning: Model class taggit.models.TaggedItem doesn't declare an explicit app_label and either isn't in an application in INSTALLED_APPS or else was imported before its application was loaded. This will no longer be supported in Django 1.9.
  class TaggedItem(GenericTaggedItemBase, TaggedItemBase):

[11/Apr/2015 10:41:26]"GET / HTTP/1.1" 404 2022
[11/Apr/2015 10:41:26]"GET /favicon.ico HTTP/1.1" 404 2055
[11/Apr/2015 10:41:34]"GET /397/ HTTP/1.1" 404 2115
[11/Apr/2015 10:41:34]"GET /favicon.ico HTTP/1.1" 404 2055
[11/Apr/2015 10:41:37]"GET /397/book_create/ HTTP/1.1" 200 3943
[11/Apr/2015 10:41:37]"GET /static/autocomplete_light/autocomplete.js HTTP/1.1" 200 21818
[11/Apr/2015 10:41:37]"GET /static/autocomplete_light/style.css HTTP/1.1" 200 4233
[11/Apr/2015 10:41:37]"GET /static/js/jquery-2.1.0.min.js HTTP/1.1" 200 83615
[11/Apr/2015 10:41:37]"GET /static/autocomplete_light/addanother.js HTTP/1.1" 200 3265
[11/Apr/2015 10:41:37]"GET /static/autocomplete_light/widget.js HTTP/1.1" 200 14632
[11/Apr/2015 10:41:37]"GET /static/autocomplete_light/remote.js HTTP/1.1" 200 1402
[11/Apr/2015 10:41:37]"GET /static/autocomplete_light/text_widget.js HTTP/1.1" 200 8597
[11/Apr/2015 10:41:37]"GET /favicon.ico HTTP/1.1" 404 2055
[11/Apr/2015 10:41:44]"GET /autocomplete/AuthorAutocomplete/?q=a HTTP/1.1" 200 52
[11/Apr/2015 10:41:44]"GET /static/autocomplete_light/xhr-pending.gif HTTP/1.1" 200 673
```

**Comment by SaeX** [comment]
Hi @jpic, was the model definition removed in `forms.py`? I've updated the test repo so it shows the issue *by default*.

I've tested in two fresh venvs with Python 2.7.9 and Python 3.4.2. I could reproduce in both environments.

Failing (note there are no Authors in the DB, if there were then I'd get a list of authors):
![Failing](https://raw.githubusercontent.com/SaeX/dal_issue_397/master/2.7.9_fails.jpg)

Working:
![Working](https://raw.githubusercontent.com/SaeX/dal_issue_397/master/2.7.9_works.jpg)

**Comment by jpic** [comment]
Could you try with 2.1.0rc3 please and close issue if fixed ?

Thanks !

**Comment by SaeX** [comment]
Sure, but can you please point me to the release I can install? I can't find 2.1.0rc3 on https://github.com/yourlabs/django-autocomplete-light/releases (goes up to 2.1.0rc2).

**Comment by jpic** [comment]
You can install with `pip install -e django-autocomplete-light==2.1.0rc3`​,
I uploaded the tag too btw.

**Comment by SaeX** [comment]
I used `pip install -e git+https://github.com/yourlabs/django-autocomplete-light.git#egg=django-autocomplete-light==2.1.0rc3`.

Confirmed fixed in 2.1.0rc3, in both venvs (2.7.9 and 3.4), thanks!

**Comment by jpic** [comment]
Awesome !!​

Thanks a lot for your feeback B)
